### PR TITLE
Refactor optimization opts

### PR DESCRIPTION
**Pull request by kripken**
Adds a nice class instead of hackish inline-c-include of a header.

### CODE REVIEW

**Comment by kripken** [comment]
I believe this is something @jfbastien suggested a while ago.

File: src/tools/optimization-options.h [hunk 1]
```diff
@@ -18,67 +18,116 @@
...
-           })
-      .add("--ignore-implicit-traps", "-iit", "Optimize under the helpful assumption that no surprising traps occur (from load, div/mod, etc.)",
-           Options::Arguments::Zero,
-           [&passOptions](Options*, const std::string&) {
-             passOptions.ignoreImplicitTraps = true;
-           })
+namespace wasm {
+
+struct OptimizationOptions : public Options {
+  const std::string DEFAULT_OPT_PASSES = "O";
```
Comment (rongjiecomputer) [inline src/tools/optimization-options.h]: static const?

Comment (kripken) [inline src/tools/optimization-options.h]: Hmm, that does sound like a good idea, but when I try it I get
````
in-class initialization of static data member ‘const string wasm::OptimizationOptions::DEFAULT_OPT_PASSES’ of non-literal type

call to non-constexpr function
````
I tried changing it to a `const char*` but it still complains about
````
error: ‘constexpr’ needed for in-class initialization of static data member ‘const char* wasm::OptimizationOptions::DEFAULT_OPT_PASSES’ of non-integral type
````

Comment (rongjiecomputer) [inline src/tools/optimization-options.h]: This works:
```
static constexpr char DEFAULT_OPT_PASSES[] = "O";
```

Comment (kripken) [inline src/tools/optimization-options.h]: Thanks, updated.

**Pull request merged**